toolboxes/feel/feelmodels/hdg/mixedpoisson.hpp

```diff
@@ -301,6 +322,75 @@ class MixedPoisson    : public ModelNumerical
+            return Feel::vf::symbolsExpr( seField, seFit, seMat );
+        }
+    constexpr auto symbolsExprField() const { return this->symbolsExprField( boost::hana::int_<Dim>() ); }
+    constexpr auto symbolsExprField( hana::int_<2> /**/ ) const
+        {
+            return Feel::vf::symbolsExpr( symbolExpr(this->keyword()+"_"+MixedPoissonPhysicsMap[M_physic]["potentialSymbol"],idv(this->potentialField()) ),
+                                          symbolExpr(this->keyword()+"_dx"+MixedPoissonPhysicsMap[M_physic]["potentialSymbol"],dxv(this->potentialField()) ),
+                                          symbolExpr(this->keyword()+"_dy"+MixedPoissonPhysicsMap[M_physic]["potentialSymbol"],dyv(this->potentialField()) ),
+                                          symbolExpr(this->keyword()+"_dn"+MixedPoissonPhysicsMap[M_physic]["potentialSymbol"],dnv(this->potentialField()) ),
+                                          symbolExpr(this->keyword()+"_"+MixedPoissonPhysicsMap[M_physic]["fluxSymbol"],idv(this->fluxField()) ),
+                                          symbolExpr(this->keyword()+"_"+MixedPoissonPhysicsMap[M_physic]["fluxSymbol"]+"x",idv(this->fluxField())(0) ),
+                                          symbolExpr(this->keyword()+"_"+MixedPoissonPhysicsMap[M_physic]["fluxSymbol"]+"y",idv(this->fluxField())(1) )
+                                          );
+        }
+    constexpr auto symbolsExprField( hana::int_<3> /**/ ) const
+        {
+            return Feel::vf::symbolsExpr( symbolExpr(this->keyword()+"_"+MixedPoissonPhysicsMap[M_physic]["potentialSymbol"],idv(this->potentialField()) ),
+                                          symbolExpr(this->keyword()+"_dx"+MixedPoissonPhysicsMap[M_physic]["potentialSymbol"],dxv(this->potentialField()) ),
+                                          symbolExpr(this->keyword()+"_dy"+MixedPoissonPhysicsMap[M_physic]["potentialSymbol"],dyv(this->potentialField()) ),
+                                          symbolExpr(this->keyword()+"_dz"+MixedPoissonPhysicsMap[M_physic]["potentialSymbol"],dzv(this->potentialField()) ),
+                                          symbolExpr(this->keyword()+"_dn"+MixedPoissonPhysicsMap[M_physic]["potentialSymbol"],dnv(this->potentialField()) ),
+                                          symbolExpr(this->keyword()+"_"+MixedPoissonPhysicsMap[M_physic]["fluxSymbol"],idv(this->fluxField()) ),
+                                          symbolExpr(this->keyword()+"_"+MixedPoissonPhysicsMap[M_physic]["fluxSymbol"]+"x",idv(this->fluxField())(0) ),
+                                          symbolExpr(this->keyword()+"_"+MixedPoissonPhysicsMap[M_physic]["fluxSymbol"]+"y",idv(this->fluxField())(1) ),
+                                          symbolExpr(this->keyword()+"_"+MixedPoissonPhysicsMap[M_physic]["fluxSymbol"]+"z",idv(this->fluxField())(2) )
+                                          );
+        }
```

Comment:
use if constexpr

Reply:
```
if constexpr ( Dim == 2 )
{}
else
{
}
```
code much easier to read and only one function

Reply:
this is the old style, in the refactoring branch, we have only 4 lines (not depends on dim or components)

Reply:
I'd rather wait until we refactor this toolbox the same way as what @vincentchabannes did in toolbox-refactoring